src/components/FilterPanel.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core";
+import { useTheme } from '@material-ui/core/styles';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Switch from "@material-ui/core/Switch";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Container from "@material-ui/core/Container";
+
+
+const useStyles = makeStyles((theme) => ({
+    formControlLabel: {
+        "text-align": "left",
+        "align-items": "start",
+        "margin-top": theme.spacing(1),
+        "width": `calc(100% - ${theme.spacing(4)}px)`
+    },
+    mdPanel: {
+        "position": "sticky",
+        "top": 0,
+        "height": "200px"
+    }
+}));
+
+export default function FilterPanel(props) {
+    const classes = useStyles();
+    const theme = useTheme();
+    const mdSize = useMediaQuery(theme.breakpoints.up('md'));
+
+    const [hasAppointments, setHasAppointments] = useState(true);
+    const [vaxType, setVaxType] = useState('Any');
+
+    const { data, onChange } = props;
+
+    useEffect(() => {
+        onChange({
+            hasAppointments: hasAppointments ?
+                d => !!d.hasAppointments === hasAppointments
```

Comment:
Isn't this function essentially the same as `d => !!d.hasAppointments`? Because it's only used when `hasAppointments` is true, so  `!!d.hasAppointments === hasAppointments` is always just checking that `!!d.hasAppointments` is true.

Also, to avoid the ternary, this whole value could be simplified to `hasAppointments: d => !hasAppointments || !!d.hasAppointments`

Reply:
Good point. This was some real galaxy brain code...

Reply:
fixed